homeassistant/components/mqtt/client.py

```diff
@@ -390,6 +394,14 @@ async def async_stop_mqtt(_event: Event) -> None:
             hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_stop_mqtt)
         )
 
+    @property
+    def subscriptions(self) -> list[Subscription]:
+        """Return the tracked subscriptions."""
+        return [
+            *chain.from_iterable(self._simple_subscriptions.values()),
+            *self._wildcard_subscriptions,
+        ]
+
```

Comment:
We can not use `mqtt_mock_entry_no_yaml_config` or `mqtt_mock_entry_with_yaml_config` in tests. Only `mqtt_mock_client` works. Possibly we should adjust the fixtures, so that this is not patched as the logic will not work. This might case tests to give unpredictable results because we expose some parts of the HA mqtt client to handle the restore.